### PR TITLE
Introduced RED.plugins to the Node-RED 1.3

### PR DESCRIPTION
**Pull request by Shaquu**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
https://nodered.org/blog/2021/04/08/version-1-3-released#node-red-plugins-framework 
https://github.com/node-red/node-red/blob/master/packages/node_modules/%40node-red/editor-client/src/js/plugins.js
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## node-red__registry ([<kbd>unpkg</kbd>](https://unpkg.com/browse/node-red__registry@latest/))
was missing the following properties:
1. init
1. load
1. clear
1. registerType
1. get

<details>
<summary>as well as these 94 other properties...</summary>
<p>registerSubflow, getNodeInfo, getNodeList, getModuleInfo, getModuleList, getNodeConfigs, getNodeConfig, getNodeIconPath, getNodeIcons, enableNode, disableNode, addModule, removeModule, installModule, uninstallModule, cleanModuleList, installerEnabled, getNodeExampleFlows, getNodeExampleFlowPath, getModuleResource, checkFlowDependencies, registerPlugin, getPlugin, getPluginsByType, getPluginList, getPluginConfigs, exportPluginSettings, deprecated, init, load, clear, registerType, get, registerSubflow, getNodeInfo, getNodeList, getModuleInfo, getModuleList, getNodeConfigs, getNodeConfig, getNodeIconPath, getNodeIcons, enableNode, disableNode, addModule, removeModule, installModule, uninstallModule, cleanModuleList, installerEnabled, getNodeExampleFlows, getNodeExampleFlowPath, getModuleResource, checkFlowDependencies, registerPlugin, getPlugin, getPluginsByType, getPluginList, getPluginConfigs, exportPluginSettings, deprecated, init, load, clear, registerType, get, registerSubflow, getNodeInfo, getNodeList, getModuleInfo, getModuleList, getNodeConfigs, getNodeConfig, getNodeIconPath, getNodeIcons, enableNode, disableNode, addModule, removeModule, installModule, uninstallModule, cleanModuleList, installerEnabled, getNodeExampleFlows, getNodeExampleFlowPath, getModuleResource, checkFlowDependencies, registerPlugin, getPlugin, getPluginsByType, getPluginList, getPluginConfigs, exportPluginSettings, deprecated</p>
</details>

## node-red__runtime ([<kbd>unpkg</kbd>](https://unpkg.com/browse/node-red__runtime@latest/))
was missing the following properties:
1. plugins
1. plugins
1. plugins
## node-red__util ([<kbd>unpkg</kbd>](https://unpkg.com/browse/node-red__util@latest/))
was missing the following properties:
1. events
1. exec
1. events
1. exec
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5b63bae3e934180d2f58eace371aa6791d727653
</p>

**Comment by tbowmo** [comment]
As I see it 98% of the changes is only formatting, which makes it difficult to locate and review the real changes. So I'm tempting to reject the PR. Unless the unrelated formatting changes could be reverted?

**Comment by Shaquu** [comment]
Also, you can hide white space changes in diff.

**Comment by tbowmo** [comment]
A lot of the changes is not whitespace, but dangling commas, and linebreaks added..

**Comment by Shaquu** [comment]
Here, check that commit
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64845/commits/db0c100a690c1d3c23976dbd45151fb3e53f00d5

File: types/node-red/index.d.ts [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
 // Project: https://github.com/node-red/node-red/tree/master/packages/node_modules/node-red, https://nodered.org/
 // Definitions by: Anders E. Andersen <https://github.com/andersea>
 //                 Thomas B. Mørch <https://github.com/tbowmo>
 //                 Bernardo Belchior <https://github.com/bernardobelchior>
 //                 Alex Kaul <https://github.com/alexk111>
+//                 Tadeusz Wyrzykowski <https://github.com/Shaquu>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// Minimum TypeScript Version: 3.1
-
-// tslint:disable:no-empty-interface
```
Comment (Shaquu) [inline types/node-red/index.d.ts:10]: Except for that line which is reverted in the other commit, all essence of PR is in that first commit.

**Comment by tbowmo** [comment]
Yeah, I just got confused about all the "extras". Wonder why they enforce a specific formatting, without reformatting the code up front..

Anyways, I had a look through the suggested commits, and can't see any dangerous things. So I'll approve it.

**Review by tbowmo — APPROVED** [review]

**Pull request merged**